### DISCUSSION TITLE
Next branch

### DISCUSSION THREAD

**Discussion by cpatti97100** [astralapp/astral#393] [General]
Hi and thanks for this fantastic project :)

Would you be so kind to list the new features we can expect on the next major release?

Thank you!

**Comment by syropian** [comment]
Hey, thanks for your interest!

In terms of the initial launch, there won't be a ton of new features, but I'll quickly outline what's changing.

- More reliable performance for people with large stars list. I haven't stress-tested it, but moving all the star caching to the user's browser storage (via IndexedDB) should be much more reliable than memcached, which was often overflowing and causing errors for people with huge amounts of starred repos.
- Polished the UI. Will still feel familiar, just lots of small improvements everywhere
- Responsive
- Dark mode
- Built on the latest and greatest — Vue 3 with TypeScript and composition API, latest version of Laravel, and Inertia. Inertia in particular has cut out tons of boilerplate. The global Vue store is now much simpler, and uses Pinia instead of Vuex. Builds and development are also much faster as it now uses Vite instead of Webpack under the hood.

#### What's next?
- I _think_ the next big feature I want to work on is public pages — I haven't fully fleshed out the idea yet, but something to allow people to share their lists with other people.
- Browser extensions
- Improved exporting, add importing
- Integration with GitHub Lists. As far as I know, GitHub does not provide API access for lists yet, but once they do I will look into how I can smoothly integrate them into Astral. 

#### Some things that are still up in the air
- Sponsor-gated features and limitations — see the section on "Making Astral Sustainable"  [here](https://github.com/astralapp/astral/discussions/340) (self-hosted will always be free)

**Reply by cpatti97100** [reply]
thank you for this overview, I look forward to it!

do you think it would also be possible to auto add the repositories "topics" as tags maybe under a special color?

It might a feature I can contribute to if you see place for it.

<img width="359" alt="image" src="https://user-images.githubusercontent.com/2529599/200015086-d0eddb2e-8643-42be-a903-e99a86805b2b.png">

**Reply by syropian** [reply]
It's definitely something I've considered. What I would probably do is add an option to show them on each repo but _not_ in the sidebar. One thing I noticed when I toyed with this idea is that it's hard to normalize topics, and a lot of them feel like duplicates of each other with slightly different spelling/punctuation, and it heavily cluttered the sidebar. Having them show on the repo should be ok, but will probably be turned **off** by default as it might create some unwanted visual clutter.

**Reply by cpatti97100** [reply]
it is definitely a good first step, I like the idea :) also, maybe the free search function can also take the topics into consideration, I think it takes into account repo title + description at the moment

**Reply by syropian** [reply]
Yep, I will probably include this as well!

BTW, it's a pretty hidden feature but the free search can filter by one or more tags as well by using a `#` to filter by tag, and a `:` as a separator for multiple tags/a search term. For example, if you wanted to find all repos that contain the word `foobar` in the title, description, or notes, with the tags `javascript` and `react` you could enter your search like so: `foobar:#javascript:#react` (order doesn't really matter).

Thanks for reminding me that I need to expose the power-search functionality somewhere 😄

**Comment by a67793581** [comment]
I'm very eager to use the new version and would like to know if the PR permission for the new version is open to everyone.
